<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>window</class>
 <widget class="QMainWindow" name="window">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>704</height>
   </rect>
  </property>
  <property name="cursor">
   <cursorShape>ArrowCursor</cursorShape>
  </property>
  <property name="windowTitle">
   <string>RSA</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QTabWidget::pane {
    border-top: 1px solid #C2C7CB;
}

QTabWidget::tab-bar {
    left: 5px;
}

QTabBar::tab {
    background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 #E1E1E1, stop: 0.4 #DDDDDD, stop: 0.5 #D8D8D8, stop: 1.0 #D3D3D3);
    border: 2px solid #C4C4C3;
    border-bottom-color: #C2C7CB;
    border-top-left-radius: 4px;
    border-top-right-radius: 4px;
    min-width: 8ex;
    padding: 3px;
}

QTabBar::tab:selected,
QTabBar::tab:hover {
    background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1, stop: 0 #fafafa, stop: 0.4 #f4f4f4, stop: 0.5 #e7e7e7, stop: 1.0 #fafafa);
}

QTabBar::tab:selected {
    border-color: #9B9B9B;
    border-bottom-color: #C2C7CB;
}

QTabBar::tab:!selected {
    margin-top: 2px;
}

QTextBrowser,
QTextEdit {
    color: rgba(0, 0, 0, 0.65);
    font-size: 14px;
    line-height: 1.5;
    background-color: #fff;
    border: 1px solid #d9d9d9;
    border-radius: 4px;
}

QLabel {
    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'PingFang SC', 'Hiragino Sans GB', 'Microsoft YaHei', 'Helvetica Neue', Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol';
    color: #0d1a26;
    font-weight: 500;
}

QPushButton {
    color: #fff;
    background-color: #1890ff;
    border-color: #1890ff;
    border: 1px solid transparent;
    border-radius: 4px;
    line-height: 1.5;
    font-weight: 400;
    white-space: nowrap;
    height: 32px;
}

QComboBox,
QAbstractItemView {
    background-color: #fff;
    border: 1px solid #d9d9d9;
    border-top-width: 1.02px;
    border-radius: 4px;
    outline: none;
    line-height: 1.5;
    font-size: 14px;
    width: 120px;
    height: 32px;
    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'PingFang SC', 'Hiragino Sans GB', 'Microsoft YaHei', 'Helvetica Neue', Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol';
    color: #0d1a26;
    font-weight: 250;
}

QComboBox::down-arrow {
    height: 12px;
    width: 12px;
    margin-right: 10px;
    image: url(:/image/down.svg);
}

QComboBox::drop-down {
    border-style: none;
}

QListView::item:hover,
QListView::item:selected {
    color: #fff;
    background: #1890ff;
}</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <spacer name="windowHorizontalSpacer">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>10</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="1" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <spacer name="leftVerticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>10</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <layout class="QVBoxLayout" name="keyVerticalLayout">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <item>
         <layout class="QHBoxLayout" name="encryptHorizontalLayout">
          <item>
           <widget class="QLabel" name="encryptionLevelLabel">
            <property name="layoutDirection">
             <enum>Qt::LeftToRight</enum>
            </property>
            <property name="styleSheet">
             <string notr="true">QLabel {
    font-size: 16px;
}</string>
            </property>
            <property name="text">
             <string>Encryption Level:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="levelComboBox">
            <property name="cursor">
             <cursorShape>ArrowCursor</cursorShape>
            </property>
            <property name="styleSheet">
             <string notr="true"/>
            </property>
            <property name="sizeAdjustPolicy">
             <enum>QComboBox::AdjustToContentsOnFirstShow</enum>
            </property>
            <item>
             <property name="text">
              <string>256 bits</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>512 bits</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>768 bits</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>1024 bits</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>2048 bits</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>4096 bits</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="generateButton">
            <property name="styleSheet">
             <string notr="true"/>
            </property>
            <property name="text">
             <string>Generate Key</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="publicKeyHorizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>13</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="publicKeyLabel">
          <property name="styleSheet">
           <string notr="true">QLabel {
    font-size: 14px;
}</string>
          </property>
          <property name="text">
           <string>Public Key</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="nLabel">
          <property name="styleSheet">
           <string notr="true">QLabel {
    font-size: 12px;
}</string>
          </property>
          <property name="text">
           <string>Value N</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextBrowser" name="nTextBrowser">
          <property name="html">
           <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'SimSun'; font-size:14px; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-family:'SimSun'; font-size:9pt;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="eLabel">
          <property name="styleSheet">
           <string notr="true">QLabel {
    font-size: 12px;
}</string>
          </property>
          <property name="text">
           <string>Value E</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextBrowser" name="eTextBrowser"/>
        </item>
        <item>
         <spacer name="privateKeyHorizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="privateKeyLabel">
          <property name="styleSheet">
           <string notr="true">QLabel {
    font-size: 14px;
}</string>
          </property>
          <property name="lineWidth">
           <number>1</number>
          </property>
          <property name="midLineWidth">
           <number>0</number>
          </property>
          <property name="text">
           <string>Private Key</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
          </property>
          <property name="margin">
           <number>0</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="pLabel">
          <property name="styleSheet">
           <string notr="true">QLabel {
    font-size: 12px;
}</string>
          </property>
          <property name="text">
           <string>Value P</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextBrowser" name="pTextBrowser"/>
        </item>
        <item>
         <widget class="QLabel" name="qLabel">
          <property name="styleSheet">
           <string notr="true">QLabel {
    font-size: 12px;
}</string>
          </property>
          <property name="text">
           <string>Value Q</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextBrowser" name="qTextBrowser"/>
        </item>
        <item>
         <widget class="QLabel" name="dLabel">
          <property name="styleSheet">
           <string notr="true">QLabel {
    font-size: 12px;
}</string>
          </property>
          <property name="text">
           <string>Value D</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextBrowser" name="dTextBrowser"/>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="messageVerticalLayout">
        <property name="leftMargin">
         <number>20</number>
        </property>
        <item>
         <widget class="QLabel" name="messageLabel">
          <property name="styleSheet">
           <string notr="true">QLabel {
    font-size: 14px;
}</string>
          </property>
          <property name="text">
           <string>Message</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextEdit" name="messageTextEdit">
          <property name="styleSheet">
           <string notr="true">QTextEdit {
    
}</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="encryptPushButton">
          <property name="cursor">
           <cursorShape>ArrowCursor</cursorShape>
          </property>
          <property name="text">
           <string>Encrypt</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="messageHorizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="encryptedMessageLabel">
          <property name="styleSheet">
           <string notr="true">QLabel {
    font-size: 14px;
}</string>
          </property>
          <property name="text">
           <string>Encrypted Message</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextEdit" name="encryptedMessageTextEdit"/>
        </item>
        <item>
         <widget class="QPushButton" name="decryptPushButton">
          <property name="text">
           <string>Decrypt</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="rightVerticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>10</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
